renormalizer/mps/tests/test_mp.py

```diff
@@ -59,3 +59,80 @@ def test_environ():
         e = complex(tensordot(l, r, axes=((0, 1, 2), (0, 1, 2)))).real
         assert pytest.approx(e) == mps.expectation(mpo)
 
+
+@pytest.mark.parametrize("comp", (True, False))
+@pytest.mark.parametrize("mp", ("mps", "mpdm", "mpo" ))
+def test_svd_compress(comp, mp):
+    
+    if mp == "mpo":
+        mps = Mpo(mol_list)
+        M = 20
+    else:
+        mps = Mps.random(mol_list, 1, 10)
+        if mp == "mpdm":
+            mps = MpDm.from_mps(mps)
+        mps.canonicalise()._dmrg_normalize()
+        M = 36
+    if comp:
+        mps = mps.to_complex(inplace=True)
+    print(f"{mps}")
+    
+    mpo = Mpo(mol_list)
+    if comp:
+        mpo = mpo.scale(-1.0j)
+    print(f"{mpo.bond_dims}")
+    
+    std_mps = mpo.apply(mps, canonicalise=True).canonicalise()
+    print(f"std_mps: {std_mps}")
+    mps.compress_config.bond_dim_max_value = M
+    mps.compress_config.criteria = CompressCriteria.fixed
+    svd_mps = mpo.contract(mps)
+    dis = svd_mps.distance(std_mps)/std_mps.dmrg_norm
+    print(f"svd_mps: {svd_mps}, dis: {dis}")
+    assert np.allclose(dis, 0.0, atol=1e-4)
+    assert np.allclose(svd_mps.dmrg_norm, std_mps.dmrg_norm, atol=1e-4)
+    
+    
+@pytest.mark.parametrize("comp", (True, False))
+@pytest.mark.parametrize("mp", ("mps", "mpdm", "mpo" ))
+def test_variational_compress(comp, mp):
+    
+    if mp == "mpo":
+        mps = Mpo(mol_list)
+        M = 20
+    else:
+        mps = Mps.random(mol_list, 1, 10)
+        if mp == "mpdm":
+            mps = MpDm.from_mps(mps)
+        mps.canonicalise()._dmrg_normalize()
+        M = 36
+    if comp:
+        mps = mps.to_complex(inplace=True)
+    print(f"{mps}")
+    
+    mpo = Mpo(mol_list)
+    if comp:
+        mpo = mpo.scale(-1.0j)
+    print(f"{mpo.bond_dims}")
+    
+    std_mps = mpo.apply(mps, canonicalise=True).canonicalise()
+    print(f"std_mps: {std_mps}")
+    
+    # 2site algorithm
+    mps.compress_config.vprocedure = [[M,1.0],[M,0.2],[M,0.1]]+[[M,0],]*10
+    mps.compress_config.vmethod = "2site"
+    var_mps = mps.variational_compress(mpo, guess=None)
+    dis = var_mps.distance(std_mps)/std_mps.dmrg_norm
+    print(f"var2_mps: {var_mps}, dis: {dis}")
+    assert np.allclose(dis, 0.0, atol=1e-4)
+    assert np.allclose(var_mps.dmrg_norm, std_mps.dmrg_norm, atol=1e-4)
+    
+    # 1site algorithm with 2site result as a guess
+    # 1site algorithm is easy to be trapped in a local minimum
```

Comment:
Is this an experience or documented in some literature?

Reply:
For the electronic system, the standard one-site algorithm usually doesn't work (see Phys. Rev. B 72, 180403(R)). The problem is widely known. 
For Holstein model, in my experience, most of the time one-site algorithm is ok but it is still not that robust compared to the two-site algorithm. In this test, you can try one-site algorithm, the result is not as accurate as two-site algorithm with the same `M`. That is to say, the mps is stuck into a local minima. 
However, since the initial state is too random in this test, I don't know if the performance will be better in a real example.
By the way, for a single mps, I guess the 2site algorithm is not much expensive than 1site algorithm in your problem. If not, I recommend to use 2site algorithm for the initial several sweeps and then 1site algorithm to converge the mps. This strategy is also used in Block (developed by Chan).

Reply:
The information is quite valuable and I think they should be put in the doc string of the method or `CompressConfig`.

Reply:
Agree.